LibreNMS/Util/Clean.php

```diff
@@ -43,6 +43,26 @@ public static function fileName($file)
         return preg_replace('/[^a-zA-Z0-9\-._]/', '', $file);
     }
 
+    /**
+     * Sanitize host name by removing all invalid characters.
+     * Behaves the same as Clean::fileName() except that it also allows valid IP addresses.
+     *
+     * @param  string  $file
+     * @return string|string[]|null
+     */
+    public static function hostName($string)
+    {
+        // If the string parses as a valid IP address it's ok
+        $address = inet_pton($string);
```

Comment:
Use `\LibreNMS\Util\IP::isValid()` instead.  In fact, maybe inline that because I plan on deleting that code anyway to fix the vulnerabilities for real.

Reply:
I could replace the `inet_pton`/`inet_ntop` pair with a call to `filter_var($string, FILTER_VALIDATE_IP)`, but that would only validate, not reformat/clean the IP address notation. Either implementation would be safe, but the current implementation cleans up the notation, which would match the name of the class better I think :)